### PR TITLE
DPL Analysis: Configurable process switches

### PR DESCRIPTION
**Pull request by aalkin**
This introduces a way to dynamically configure which process functions a task will run.
* Added a `ProcessConfigurable` with a distinct kind to be used with hyperloop
* Added a macro to declare process configurables in a task
* Added workflow adjustment from supplied configuration (WARN: only JSON configuration is supported currently, commandline switches will not work)
* Updated multiple process tutorial example

### CODE REVIEW

File: Analysis/Tutorials/src/multiProcess.cxx [hunk 1]
```diff
@@ -63,48 +63,41 @@ struct MultipleProcessExample {
...
+  void processRec(soa::Filtered<aod::Collisions>::iterator const&, aod::Tracks const& tracks)
   {
     for (auto& track : tracks) {
       registry.fill(HIST("etaRec"), track.eta());
       registry.fill(HIST("phiRec"), track.phi());
     }
   }
 
-  void processGen(soa::Filtered<aod::McCollisions>::iterator const& mcCollision, aod::McParticles const& mcParticles)
+  PCONF(prec, "prec", "Process reco level", &MultipleProcessExample::processRec, true);
```
Comment (jgrosseo) [inline Analysis/Tutorials/src/multiProcess.cxx]: Let's put a longer name `PROCESS_CONFIGURABLE` or smth better?

**Review by jgrosseo — COMMENTED** [review]
Looks good to me! I'd change the name of the macro though.

File: Analysis/Tutorials/src/multiProcess.cxx (hunk 1 above)
Comment (aalkin) [reply Analysis/Tutorials/src/multiProcess.cxx]: Sure, this is was a proof-of-concept at first and I did not change it. It is still required to declare this configurable after the process function it refers to, as it needs the pointer.

**Comment by aalkin** [comment]
Marking this ready for review.

Note that changes in tasks are made to preserve current functionality where the process selection is controlled by a a workflow option. Analyzers can later improve that by using the process switches directly.

**Review by ktf — DISMISSED** [review]

**Review by jgrosseo — DISMISSED** [review]

**Comment by jgrosseo** [comment]
@aalkin 
I don't think this can be related to your PR, can it?
` 20/383 Test  #17: Framework/Utils/test/test_RootTreeWriterWorkflow.cxx ...........................Subprocess aborted***Exception:   1.05 sec
[INFO] Initialising O2 Data Processing Layer. Driver PID: 15525.
[INFO] O2 Data Processing Layer initialised. We brake for nobody.
[INFO] No device resulting from the workflow. Quitting.
Error in <TFile::TFile>: file /tmp/test_RootTreeWriter.root does not exist
Error in <TFile::TFile>: file /tmp/test_RootTreeWriter_alt.root does not exist
terminate called after throwing an instance of 'std::runtime_error'
  what():  missing file /tmp/test_RootTreeWriter.root
`

**Comment by aalkin** [comment]
Not sure, I will check.

**Comment by aalkin** [comment]
@jgrosseo seems to be genuine error, I'll look into it

**Review by jgrosseo — APPROVED** [review]

**Comment by ktf** [comment]
This seems to correlate with failing train runs since the 31st. Is it being looked into?

**Comment by aalkin** [comment]
> This seems to correlate with failing train runs since the 31st. Is it being looked into?

Yes, I missed to add an exclusion, testing the fix right now.

**Pull request merged**